### ISSUE TITLE
TypeError: can only concatenate str (not "bytes") to str

### ISSUE THREAD

**Issue by cjpark-sapcsa** [microsoft/autogen#17]
Hi Team, 

try some notebook in windows VSC.. i got the below error. Anyone seeing before?

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[c:\My](file:///C:/My) Work\17. AI learning\autogen\notebook\agentchat_groupchat_vis.ipynb Cell 10 line 1
----> [1](vscode-notebook-cell:/c%3A/My%20Work/17.%20AI%20learning/autogen/notebook/agentchat_groupchat_vis.ipynb#X12sZmlsZQ%3D%3D?line=0) user_proxy.initiate_chat(manager, message="download data from https://raw.githubusercontent.com/uwdata/draco/master/data/cars.csv and plot a visualization that tells us about the relationship between weight and horsepower. Save the plot to a file. Print the fields in a dataset before visualizing it.")
      [2](vscode-notebook-cell:/c%3A/My%20Work/17.%20AI%20learning/autogen/notebook/agentchat_groupchat_vis.ipynb#X12sZmlsZQ%3D%3D?line=1) # type exit to terminate the chat

File [c:\My](file:///C:/My) Work\17. AI learning\autogen\.conda\lib\site-packages\autogen\agentchat\conversable_agent.py:521, in ConversableAgent.initiate_chat(self, recipient, clear_history, silent, **context)
    507 """Initiate a chat with the recipient agent.
    508 
    509 Reset the consecutive auto reply counter.
   (...)
    518         "message" needs to be provided if the `generate_init_message` method is not overridden.
    519 """
    520 self._prepare_chat(recipient, clear_history)
--> 521 self.send(self.generate_init_message(**context), recipient, silent=silent)

File [c:\My](file:///C:/My) Work\17. AI learning\autogen\.conda\lib\site-packages\autogen\agentchat\conversable_agent.py:324, in ConversableAgent.send(self, message, recipient, request_reply, silent)
    322 valid = self._append_oai_message(message, "assistant", recipient)
    323 if valid:
--> 324     recipient.receive(message, self, request_reply, silent)
    325 else:
    326     raise ValueError(
    327         "Message can't be converted into a valid ChatCompletion message. Either content or function_call must be provided."
    328     )
...
--> 903 logs_all += "\n" + logs
    904 if exitcode != 0:
    905     return exitcode, logs_all

TypeError: can only concatenate str (not "bytes") to str
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?0b0f12a6-e386-4665-8bc2-b2edb08dedb9) or open in a [text editor](command:workbench.action.openLargeOutput?0b0f12a6-e386-4665-8bc2-b2edb08dedb9). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

**Comment by sonichi** [comment]
Is pyautogen version 0.1.1 being used?

**Comment by cjpark-sapcsa** [comment]
Hi @sonichi   

(c:\My Work\17. AI learning\autogen\.conda) PS C:\My Work\17. AI learning\autogen\notebook> pip show pyautogen
Name: pyautogen
Version: 0.1.1
Summary: Enabling Next-Gen LLM Applications via Multi-Agent Conversation Framework   
Home-page: https://github.com/microsoft/autogen
Author: AutoGen
Author-email: auto-gen@outlook.com
License: UNKNOWN
Location: c:\my work\17. ai learning\autogen\.conda\lib\site-packages
Requires: diskcache, flaml, openai, termcolor
Required-by:
(c:\My Work\17. AI learning\autogen\.conda) PS C:\My Work\17. AI learning\autogen\notebook>

**Comment by sonichi** [comment]
This problem should have been fixed in v0.1.1. @rickyloynd-microsoft have you tried this notebook in windows?

**Comment by rickyloynd-microsoft** [comment]
No, but I'll try it tomorrow.

**Comment by S0ulfire84** [comment]
I have the same issue.

This is in Python in Windows. Python 3.11, Autogen 0.1.1

 pip show pyautogen
Name: pyautogen
Version: 0.1.1
Summary: Enabling Next-Gen LLM Applications via Multi-Agent Conversation Framework
Home-page: https://github.com/microsoft/autogen
Author: AutoGen
Author-email: auto-gen@outlook.com
License: UNKNOWN
Location: C:\Python311\Lib\site-packages
Requires: diskcache, flaml, openai, termcolor
Required-by:
PS C:\...\AutogenTestDrive> python --version             
Python 3.11.0

**Comment by LittleLittleCloud** [comment]
Also hit the same issue

The bug should be here
https://github.com/microsoft/autogen/blob/0dff1237c17505d9d072940a500f3e93c515e60e/autogen/code_utils.py#L288

where
```
# stderr's type is byte!!!
logs = result.stderr
```

The fix can be simply decode the byte
```
logs = result.stderr.decode('utf-8')
```

**Comment by petercoulton** [comment]
> also line 296? Is this a windows-specific issue? The decoding is not needed for linux.

I'm getting the same error with pyautogen-0.1.1 on Windows, so could be.

**Comment by geg00** [comment]
After fixing lines 288 and 296 reinstalled locally and now I'm getting this error

File [~\AppData\Local\miniforge3\envs\kase-webapp\Lib\subprocess.py:1493](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/hbb9279/github/autogen/notebook/~/AppData/Local/miniforge3/envs/kase-webapp/Lib/subprocess.py:1493), in Popen._execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, unused_restore_signals, unused_gid, unused_gids, unused_uid, unused_umask, unused_start_new_session, unused_process_group)
   1491 # Start the process
   1492 try:
-> 1493     hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
   1494                              # no special security
   1495                              None, None,
   1496                              int(not close_fds),
   1497                              creationflags,
   1498                              env,
   1499                              cwd,
   1500                              startupinfo)
   1501 finally:
   1502     # Child is launched. Close the parent's copy of those pipe
   1503     # handles that only the child should have open.  You need
   (...)
   1506     # pipe will not close when the child process exits and the
   1507     # ReadFile will hang.
   1508     self._close_pipe_fds(p2cread, p2cwrite,
   1509                          c2pread, c2pwrite,
   1510                          errread, errwrite)
FileNotFoundError: [WinError 2] The system cannot find the file specified

**Comment by LittleLittleCloud** [comment]
Looks like there's a bug on running python code using docker on windows . Can you try

Pip install docker

To install docker dependency and check if the error goes away? The docker dependency is somehow missing in setup.py and that might be the cause?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by rickyloynd-microsoft** [comment]
I followed the installation steps in the win branch (`pip install pyautogen`), then modified and renamed OAI_CONFIG_LIST_sample.json as suggested, and ran the new test (`python test/twoagent.py`). The first agent turn replied with code as expected. I pressed enter to use auto-reply, and got the error mentioned in  issue #17:

`pywintypes.error: (2, 'CreateFile', 'The system cannot find the file specified.')`

Then I did `pip install docker` and ran the test again. The error changed:

`docker.errors.DockerException: Error while fetching server API version: (2, 'CreateFile', 'The system cannot find the file specified.')`
